pkg/channel/consolidated/dispatcher/dispatcher.go

```diff
@@ -93,6 +94,29 @@ func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispat
 	conf.Consumer.Return.Errors = true    // Returns the errors in ConsumerGroup#Errors() https://godoc.org/github.com/Shopify/sarama#ConsumerGroup
 	conf.Producer.Return.Successes = true // Must be enabled for sync producer
 
+	// Get the auth info
+	if args.AuthSecretName != "" {
+		kafkaAuthCfg := utils.GetKafkaAuthData(ctx, args.AuthSecretName, args.AuthSecretNamespace)
```

Comment:
Do we have to perform this resolution in the dispatcher? Can't we do it at controller level?

Reply:
@slinkydeveloper I moved